asr1k_neutron_l3/plugins/l3/agents/asr1k_l3_agent.py

```diff
@@ -724,9 +733,14 @@ def _process_routers_loop(self):
             LOG.warning("The processing thread pool size has been reduced to match 'yang_connection_pool_size' "
                         "its now {}".format(poolsize))
 
-        pool = eventlet.GreenPool(size=poolsize)
-        while True:
-            pool.spawn_n(self._process_router_update)
+        with ThreadPoolExecutor(max_workers=poolsize) as executer:
+            while not self._stop:
+                # the executer worker queue is infinite and doesn't block
+                if self._queue.get_size():
+                    executer.submit(self._process_router_update)
+                else:
+                    time.sleep(5)
```

Comment:
`5` looks like some magic value. Should we add some comments here?

Reply:
Yes, it's very magic. I already commented above that the worker queue is infinite, so we just wait a bit before checking for new queue items. The values is a quite arbitrary small value. Any suggestion?

Reply:
Let's leave this for now. I don't think that we will need to control this period. in the near future.